src/java.base/share/classes/java/lang/invoke/MethodType.java

```diff
@@ -1223,15 +1224,16 @@ static MethodType fromDescriptor(String descriptor, ClassLoader loader)
      * Returns a descriptor string for the method type.  This method
      * is equivalent to calling {@link #descriptorString() MethodType::descriptorString}.
      *
-     * <p>
-     * Note that this is not a strict inverse of {@link #fromMethodDescriptorString fromMethodDescriptorString}.
+     * @apiNote
+     * This is not a strict inverse of {@link #fromMethodDescriptorString fromMethodDescriptorString}
+     * because the latter requires a suitable class loader argument.
      * Two distinct classes which share a common name but have different class loaders
      * will appear identical when viewed within descriptor strings.
+     * In addition, the result string is not a bytecode descriptor if this
+     * method type cannot be {@linkplain ##descriptor described nominally}.
```

Comment:
This method is equivalent to `MethodType::descriptorString` which specifies the result descriptor string if a method type descriptor not.  So I think this  sentence may not be needed.

Reply:
Should we consolidate this distinct class loader note into `descriptorString` too?

Reply:
This method was added prior to `MethodTypeDesc`.    MethodTypeDesc::ofDescriptor` and `MethodTypeDesc::resolveConstantDesc` would be the alternative way to get back the method type.  I tend to think that `descriptorString` would not need that distinct class loader note (as it would use the Lookup class to resolve instead).

Reply:
Since the method type descriptor info is available in `descriptorString` but it's a reason `toMethodDescriptorString` is not a strict inverse of `fromMethodDescriptorString`, I propose to reword the notes sections to the following:
```
     * @apiNote
     * This is not a strict inverse of {@link #fromMethodDescriptorString
     * fromMethodDescriptorString} which requires a method type descriptor
     * (JVMS {@jvms 4.3.3}) and a suitable class loader argument. Two distinct
     * classes which share a common name but have different class loaders will
     * appear identical when viewed within descriptor strings.
     * <p>
     * This method is included for the benefit of applications that must
     * generate bytecodes that process method handles and {@code invokedynamic}.
```

Reply:
That's good.